### PR TITLE
Strings updates

### PR DESCRIPTION
**Pull request by chloepugh**
Condenses the strings chapter to better fit in the allotted time, and adds a new exercise that should be easier to complete in the time given

### CODE REVIEW

**Comment by ian-rickard** [comment]
```{r, purl=purl_example_code}
# Collasping a character vector
vector_to_collapse <- c("some", "strings", "to", "combine")
stringr::str_c(vector_to_collapse, collapse="")
```

I would have thought it could be more illustrative to use `collapse=" "`

also - typo in the annotation

File: rmd_files/strings.Rmd [hunk 1]
```diff
@@ -3,565 +3,268 @@
...
-
-```{r, include=show_solution, purl=purl_solutions}
-stringr::str_subset(words, "^[aeiou].*[^aeiou]$") %>% head()
-#> [1] "about"  "accept"  "account" "across"  "act"  "actual"
-start_with_vowel <- stringr::str_detect(words, "^[aeiou]")
-end_with_consonant <- stringr::str_detect(words, "[^aeiou]$")
-words[start_with_vowel & end_with_consonant] %>% head()
-#> [1] "about"  "accept"  "account" "across"  "act"  "actual"
-
+Or only strings that contain letters and numbers:
```
Comment (ian-rickard) [inline rmd_files/strings.Rmd:194]: maybe 'either letters or numbers'

File: rmd_files/strings.Rmd [hunk 2]
```diff
@@ -3,565 +3,268 @@
...
 
-```{r, include=show_solution, purl=purl_solutions}
-
-unique(unlist(stringr::str_extract_all(sentences, "\\b[A-Za-z]{3,}s\\b"))) %>%
-  head()
-#> [1] "planks" "days"   "bowls"  "lemons" "makes"  "hogs"
+```{r, purl=purl_example_code}
+colours <- c("scarlet...red", "ultramarine.blue", "cadmium_red", "cobalt blue", "cerulean-blue")
 
+# Replace the first character that isn't a letter or number with an underscore
```
Comment (ian-rickard) [inline rmd_files/strings.Rmd:245]: This returns

```
[1] "scarlet_..red"    "ultramarine_blue" "cadmium_red"      "cobalt_blue"     
[5] "cerulean_blue"   
[1] "scarlet___red"    "ultramarine_blue" "cadmium_red"      "cobalt_blue"     
[5] "cerulean_blue"  
```

Actually my regex isn't good enough for me to be able to suggest the solution, so I leave it to you!

File: rmd_files/strings.Rmd (hunk 1 above)
Comment (chloepugh) [reply rmd_files/strings.Rmd]: Fixed

File: rmd_files/strings.Rmd (hunk 2 above)
Comment (chloepugh) [reply rmd_files/strings.Rmd:247]: As discussed - this was just to demonstrate the difference in behaviour between `str_replace` and `str_replace_all`, although could possibly think of some better example data...

I've added an extra sentence underneath to point out what's happened.

**Pull request merged**